dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/config/TrackerValidationConfig.java

```diff
@@ -50,60 +72,77 @@
 @Configuration( "trackerImportValidationConfig" )
 public class TrackerValidationConfig
 {
-    @Bean( "ruleEngineValidationHooks" )
-    public List<Class<? extends TrackerValidationHook>> getRuleEngineValidationHooks()
+    private final Map<Class<? extends TrackerValidationHook>, TrackerValidationHook> validationHooks;
+
+    public TrackerValidationConfig( Collection<TrackerValidationHook> checkers )
+    {
+        validationHooks = byClass( checkers );
+    }
+
+    @SuppressWarnings( "unchecked" )
+    private <T> Map<Class<? extends T>, T> byClass( Collection<T> items )
```

Comment:
Does this needs to be generic?
Can we just substitute T with TrackerValidationHook?

Reply:
I think it should be generic if it does a generic operation otherwise you miss to see that this might even be elevated to a general utility class. Also signatures become longer and harder to ready. Using the generic helps to understand that we are doing a general operation where the type does not matter.

Reply:
hopefully resolved 😋